File: src/ca/mindmagic/game/map/grid/shape/Square.java
package ca.mindmagic.game.map.grid.shape;

import java.awt.Point;

public class Square extends Shape {

  public Square(){
  }

  public Square(int sideLength){
    if(sideLength > 0){
      this.sideLength = sideLength;
    }
  }

  @Override public int numberOfSides() {
    return 4;
  }

  @Override public int lengthOfSides() {
    return sideLength;
  }

  @Override public Point[] getVertices() {
    return getVertices(sideLength);
  }

  public static Point[] getVertices(int sideLength){
    Point[] vertices = new Point[4];
    int half = sideLength/2;
    vertices[0] = new Point(-half, -half);
    vertices[1] = new Point(half, -half);
    vertices[2] = new Point(half, half);
    vertices[3] = new Point(-half, half);
    return vertices;
  }
}

File: src/ca/mindmagic/game/map/grid/pattern/square/Square.kt
package ca.mindmagic.game.map.grid.pattern.square

import ca.mindmagic.game.map.grid.pattern.Shape
import javafx.geometry.Point2D

class Square(sideLength: Double) : Shape(sideLength) {


    override fun numberOfSides(): Int {
        return 4
    }

    override fun getVertices(): Array<Point2D> {
        return getVertices(sideLength)
    }

    companion object {

        fun getVertices(sideLength: Double): Array<Point2D> {
            val half = sideLength / 2.0
            return arrayOf(
                    topLeft(half),
                    topRight(half),
                    bottomRight(half),
                    bottomLeft(half)
            )

        }

        private fun topLeft(half: Double) = Point2D(-half, -half)
        private fun topRight(half: Double) = Point2D(half, -half)
        private fun bottomRight(half: Double) = Point2D(half, half)
        private fun bottomLeft(half: Double) = Point2D(-half, half)
    }
}
